### PR TITLE
[FrameworkBundle] grab a service from the container only if it exists

### LINKED ISSUE #53551: `cache:pool:clear --all` command fails in a prod environment
### Symfony version(s) affected

7.0.2, 6.4.2

### Description

When trying to clear cache pools using the `cache:pool:clear --all` command while in a production environment (i.e. `APP_ENV=prod`), the command fails with the following error on Symfony 6.4:

```
In Container.php line 249:
                                                                                                                                                                                                              
  The "cache.annotations" service or alias has been removed or inlined when the container was compiled. You should either make it public, or stop using the container directly and use dependency injection   
  instead.                                                                                                                                                                                                    
                                                                                                                                                                                                              

cache:pool:clear [--all] [--exclude EXCLUDE] [--] [<pools>...]

Failed to execute command bin/console cache:pool:clear --all: exit status 1
```

And on Symfony 7.0:

```
In Container.php line 240:
                                                                                                                                                                                                               
  The "cache.asset_mapper" service or alias has been removed or inlined when the container was compiled. You should either make it public, or stop using the container directly and use dependency injection   
  instead.                                                                                                                                                                                                     
                                                                                                                                                                                                               

cache:pool:clear [--all] [--exclude EXCLUDE] [--] [<pools>...]

Failed to execute command bin/console cache:pool:clear --all: exit status 1
```

Both of these happen only in the prod env, the dev env works perfectly fine.

### How to reproduce

1. Create a new webapp installation of the app from `symfony/skeleton`.
2. Change `APP_ENV` to `prod`.
3. Attempt to run the `cache:pool:clear --all` command.

### Possible Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix #53551
| License       | MIT

### CODE REVIEW

**Comment by xabbuh** [comment]
The root cause for the observed behaviour is that the list of pool names passed to the constructor of `CachePoolClearCommand` command contains also the ids of cache pools that are later cleaned up by the `RemoveUnusedDefinitionsPass`. An alternative solution would be to collect available cache pool names at a later stage during compilation when the cleanup of unused service definitions has already happened.

**Comment by nicolas-grekas** [comment]
Can't we check if the service exists before accessing it? Because this strategy is going to keep around pools that are not used, preventing them to be cleaned up by the compilation process.

**Comment by xabbuh** [comment]
indeed, that's a better idea

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Why the Validator label?

**Comment by xabbuh** [comment]
probably because I made a typo in the initial title and didn't remove the label when fixing that so Carson re-adds it again

File: src/Symfony/Bundle/FrameworkBundle/Command/CachePoolClearCommand.php [hunk 1]
```diff
@@ -70,7 +70,9 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $clearers = [];
 
         $poolNames = $input->getArgument('pools');
+        $clearAll = false;
         if ($input->getOption('all')) {
+            $clearAll = true;
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Command/CachePoolClearCommand.php]:
```suggestion
        if ($clearAll = $input->getOption('all')) {
```

**Review by nicolas-grekas — APPROVED** [review]
LGTM with one minor tweak

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**